core/src/main/resources/jenkins/model/Jenkins/login.jelly

```diff
@@ -23,58 +23,120 @@ THE SOFTWARE.
+      <meta name="ROBOTS" content="NOFOLLOW" />
+      <!-- css styling, will fallback to default implementation -->
+      <st:include it="${loginDecorator}" page="login-css.jelly" optional="true"/>
+    </head>
+    <body>
+      <j:choose>
+        <j:when test="${it.setupWizard.isUsingSecurityToken()}">
+          <st:include it="${it.setupWizard}" page="authenticate-security-token"/>
+        </j:when>
+        <j:otherwise>
+          <div class="login-main">
+            <div class="modal">
+              <st:include it="${loginDecorator}" page="login-header.jelly" optional="true"/>
+              <!-- login form -->
+              <form name="login" action="${it.securityRealm.authenticationGatewayUrl}" method="post">
+                <j:if test="${it.securityRealm.allowsSignup()}">
+                  <div style="margin-top:2em">
+                    ${%signUp}
+                  </div>
+                </j:if>
+                <j:if test="${error}">
+                  <div class="alert alert-danger">
+                    ${%Invalid username or password}
+                  </div>
+                </j:if>
+                <div>
+                  <input
+                    class="${inputClass}"
+                    type="text"
+                    name="j_username"
+                    id="j_username"
+                    autocorrect="off"
+                    autocapitalize="off"
+                    placeholder="${%User}"
+                  />
+                </div>
+                <div>
+                  <input
+                    class="${inputClass}"
+                    type="password"
+                    name="j_password"
+                    placeholder="${%Password}"
+                  />
+                </div>
+                <input type="hidden" name="from" value="${from}" />
+                <div class="submit"><f:submit value="${%signIn}" /></div>
+                <!-- focus the username field -->
+                <script type="text/javascript">
+                  document.getElementById('j_username').focus();
+                  var checkBoxClick = function(event) {
+                    document.getElementById('remember_me').click();
+                  }
+                </script>
 
-        <j:forEach var="fls" items="${it.getFederatedLoginServices()}">
-          <div>
-            <st:include page="loginFragment.jelly" it="${fls}"/>
-          </div>
-        </j:forEach>
+                <j:if test="${app.security.name()=='SECURED' and !app.disableRememberMe}">
+                  <!-- remember me service only available for Acegi -->
+                  <div class="checkboxWrapper">
+                    <div class="checkbox">
+                      <input type="checkbox" id="remember_me" name="remember_me" />
+                      <label for="remember_me"> </label>
+                    </div>
+                    <span onClick="checkBoxClick()">${%Keep me signed in}</span>
```

Comment:
Why is this done with JS and the label left empty?

Reply:
we use the label for graphical purposes.

Reply:
I don't understand what this means.

Isn't this inaccessible?

Reply:
Well the label is only for mouse clicks and not for the ' tab' focus

> This is mentioned in the HTML 4.01 Ref here: http://www.w3.org/TR/html401/interact/forms.html#h-17.9.1
> Specifically near the end of section 17.9.1 and just before 17.10:

    When a LABEL element receives focus, it passes the focus on to its associated control.

Reply:
AFAIR, clicking labels (un)checks the associated text box. So why aren't we using the label as it is intended to be used, instead using this JS hack?

Reply:
We implemented the same behavior of clicking the label via the onClick method. The styling of the checkbox needed some "dom" tricks to get it working with css. However if that is a big problem for you I will drop the styling of checkbox.